recipes/onnxruntime/all/patches/1.14.1-0001-cmake-dependencies.patch

```diff
@@ -0,0 +1,244 @@
+--- a/cmake/CMakeLists.txt
++++ b/cmake/CMakeLists.txt
+@@ -753,7 +753,7 @@ endif()
+ 
+ function(onnxruntime_set_compile_flags target_name)
+     if (CPUINFO_SUPPORTED)
+-      onnxruntime_add_include_to_target(${target_name} cpuinfo)
++      onnxruntime_add_include_to_target(${target_name} cpuinfo::cpuinfo)
+     endif()
+     if(onnxruntime_ENABLE_EAGER_MODE)
+       target_compile_definitions(${target_name} PRIVATE ENABLE_EAGER_MODE)
+@@ -837,7 +837,7 @@ function(onnxruntime_set_compile_flags target_name)
+         target_compile_options(${target_name} PRIVATE "-Wno-unused-parameter")
+       endif()
+       target_compile_definitions(${target_name} PUBLIC -DNSYNC_ATOMIC_CPP11)
+-      target_include_directories(${target_name} PRIVATE "${google_nsync_SOURCE_DIR}/public")
++      onnxruntime_add_include_to_target(${target_name} nsync::nsync_cpp)
+     endif()
+     foreach(ORT_FLAG ${ORT_PROVIDER_FLAGS})
+       target_compile_definitions(${target_name} PRIVATE ${ORT_FLAG})
+@@ -1461,7 +1461,7 @@ if (WIN32)
+   list(APPEND onnxruntime_EXTERNAL_LIBRARIES ${SYS_PATH_LIB})
+   list(APPEND onnxruntime_EXTERNAL_LIBRARIES debug Dbghelp)
+ else()
+-  list(APPEND onnxruntime_EXTERNAL_LIBRARIES nsync_cpp)
++  list(APPEND onnxruntime_EXTERNAL_LIBRARIES nsync::nsync_cpp)
+   list(APPEND onnxruntime_EXTERNAL_LIBRARIES ${CMAKE_DL_LIBS} Threads::Threads)
+ endif()
+ 
+--- a/cmake/external/onnxruntime_external_deps.cmake
++++ b/cmake/external/onnxruntime_external_deps.cmake
+@@ -237,7 +237,10 @@ if (NOT WIN32)
+   #nsync tests failed on Mac Build
+   set(NSYNC_ENABLE_TESTS OFF CACHE BOOL "" FORCE)
+   onnxruntime_fetchcontent_makeavailable(google_nsync)
+-  set(nsync_SOURCE_DIR ${google_nsync_SOURCE_DIR})
++  if(google_nsync_SOURCE_DIR)
++    add_library(nsync::nsync_cpp ALIAS nsync_cpp)
++    target_include_directories(nsync_cpp PUBLIC ${google_nsync_SOURCE_DIR}/public)
++  endif()
+ endif()
+ 
+ if(onnxruntime_USE_CUDA)
+@@ -361,6 +364,9 @@ FetchContent_Declare(
+ 
+ if (CPUINFO_SUPPORTED)
+   onnxruntime_fetchcontent_makeavailable(pytorch_cpuinfo)
++  if(pytorch_cpuinfo_SOURCE_DIR)
++    add_library(cpuinfo::cpuinfo ALIAS cpuinfo)
```

Comment:
From the CMakeDeps generator you can override these to match what the consumer wants so it's less patching

Reply:
everything in this file has been merged upstream so I'd rather not changed it.
It might be interesting for 1.14.1-0002-cmake-dependencies.patch if you have a specific example to work from.